docs/misc/commands.md

```diff
@@ -0,0 +1,122 @@
+Here is an example:
+```java
+public static void register(CommandDispatcher<CommandSourceStack> dispatcher) {
+    dispatcher.register (
+        Commands.literal("commandname")
+        .then(Commands.argument("message", StringArgumentType.string())) // or word(), it's the string type
+        .execute(context -> {
+            String message = StringArgumentType.getString(context, "message"); // Same name as above
+            ServerPlayer player = context.getSource(); // Who is executing the command
+            player.sendServerMessage(message);
+            return Command.SINGLE_SUCCES;
+        })
+    );
+}
+```
+:::danger
+Note that all ArgumentTypes are abstract and you need to use child, as `#string()` in this example.
+:::
+### Suggestion
+You can suggest (implement auto-completion) things with method `#suggest(SuggestionProvider)` just after a `#then` expression.
+A `SuggestionPovider` take this form:
+```java
+// A list of suggestion
+private static final List<String> ACTIONS = List.of("add", "sub", "set");
+
+private static final SuggestionProvider<CommandSourceStack> ACTION_SUGGESTIONS = (context, builder) ->
+    SharedSuggestionProvider.suggest(ACTIONS, builder); // We use ACTIONS list
+```
+:::note
+Users aren't obliged to use suggestions. If you want to error when gived argument doesn't match with suggestion, you can do in the execute statement:
+```java
+if(!ACTIONS.contains(arg)) { // Arg is already declared
+    context.getSource().sendFailure(Component.literal("Invalid action (add/sub/set)")); // As return send_succes method, but it send failure (red message)
+    return 0; // Ensure to stop the function
+}
+```
+:::
+## Other features
+As many Neoforge elements, command has his own set of function that make our life better...
+### Execution conditions
+You can use the `#requires` method after `Commands.literal` function to require some things with based on an function that take one argument: `context`.
+```java
+// In the register function
+dispatcher.register(
+    Commands.literal("opcommand")
+        .requires(context -> context.hasPermission(2))
+);
+```
+:::note
+Of course, you can place `#requires` in a `if` statement or only when certains parameters equals to somethings.
+:::
+### Side requirement
+In your `register` method, you have a parameter `dispatcher`. In addition to enabling registration, it gave some informations about tick game context. One of these information is `#getCommandSelection`, which returns whether it is multiplayer or singleplayer.
+```java
+public static void register(CommandDispatcher<CommandSourceStack> dispatcher) {
+        if(event.getCommandSelection() == Commands.CommandSelection.DEDICATED) { // INTEGRATED for singleplayer
+            dispatcher.register(
+                Commands.literal("multiplayercommand")
+                        // then code action
+            );
+        }
+    }
+```
+### `Commands.literal`
```

Comment:
Are you trying to explain the subcommand structure? What does this have to do with `Commands#literal`?  This goes back to the `ArgumentBuilder`.